src/cluster.c

```diff
@@ -4155,30 +4155,38 @@ void clusterCron(void) {
         clusterUpdateState();
 }
 
+/* Returns 1 if the client is currently using a plaintext connection in a TLS
+ * cluster, 0 otherwise. (Vice versa is not possible, but it might be in the
+ * future.) */
+static inline int clusterClientUsesAltPort(client *c) {
+    return server.tls_cluster && c->conn &&
+        connGetType(c->conn) != CONN_TYPE_TLS;
+}
+
+/* For redirects, verb must start with a dash, e.g. "-ASK" or "-MOVED". */
```

Comment:
I did it first, but we also need with dash for redirects (resp errors) and without dash for pubsub. Flags maybe? I think a string like this is at least simple.

Reply:
Flags were what I thought as well. That said, I don't have a strong opinion here since I don't see the use case of this function getting expanded further and we have tests to validate the response/message strings.